### PR TITLE
Team_D_project_changed_Plane.lean_and_Ray.lean

### PR DESCRIPTION
**Pull request by kfc2333**

### CODE REVIEW

File: EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean [hunk 1]
```diff
@@ -47,35 +47,75 @@ instance : AddTorsor (Vec) P := by infer_instance
 
 /- vector $AB +$ point $A =$ point $B$ -/
 @[simp]
-theorem start_vadd_vec_eq_end (A B : P) : (VEC A B) +ᵥ A = B := sorry
+theorem start_vadd_vec_eq_end (A B : P) : (VEC A B) +ᵥ A = B := by
+  rw [Vec.mk_pt_pt]
+  exact vsub_vadd B A
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean:52]: Here `rw [Vec.mk_pt_pt]` only unfold the definition. You can directly write `:= by exact vsub_vadd B A` to show this theorem, or even shorter `:= vsub_vadd B A`.

`simp` do need a `rw` to unfold the definition. But `exact` don't need it.

File: EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean [hunk 2]
```diff
@@ -47,35 +47,75 @@ instance : AddTorsor (Vec) P := by infer_instance
...
+  have k : v +ᵥ A -ᵥ A = A-ᵥ A := by
+    rw [h]
+  have u : v +ᵥ A -ᵥ A = v := by
+    simp
+  rw [u] at k
+  simp at k
+  exact k
+  intro h
+  rw[h]
+  simp
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean:67]: It is possible to write this proof in a shorter way. Please read https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/AddTorsor.html# and find the correct theorems. One way is exactly `zero_vadd`. the other way can be also reduced in length using correct theorems.

File: EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean [hunk 3]
```diff
@@ -47,35 +47,75 @@ instance : AddTorsor (Vec) P := by infer_instance
...
 
 @[simp]
-theorem vec_same_eq_zero (A : P) : VEC A A = 0 := sorry
+theorem vec_same_eq_zero (A : P) : VEC A A = 0 := by
+  rw [Vec.mk_pt_pt]
+  simp
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean:72]: Please use `simp?` to change `simp` into `simp only` in all places.

File: EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean [hunk 4]
```diff
@@ -47,35 +47,75 @@ instance : AddTorsor (Vec) P := by infer_instance
...
 theorem pt_eq_pt_of_eq_smul_smul {O A B : P} {v : Vec} {tA tB : ℝ} (h : tA = tB) (ha : VEC O A = tA • v) (hb : VEC O B = tB • v) : A = B := by
   have hc : VEC A B = VEC O B - VEC O A := Eq.symm (vec_sub_vec O A B)
   rw [ha, hb, ← sub_smul, Iff.mpr sub_eq_zero (Eq.symm h), zero_smul] at hc
   exact Eq.symm ((eq_iff_vec_eq_zero A B).2 hc)
 
 theorem eq_of_smul_Vec_nd_eq_smul_Vec_nd {v : Vec_nd} {tA tB : ℝ} (e : tA • v.1 = tB • v.1) : tA = tB := by
-  sorry
+  have h : (tA - tB) • v.1 = 0 := by
+    rw [sub_smul]
+    rw [e]
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Basic/Plane.lean:111]: if you `rw` many theorems, you can write them in one line. `rw [a, b]`

File: EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean [hunk 5]
```diff
@@ -257,19 +292,70 @@ section existence
...
+theorem Seg_nd.exist_pt_beyond_pt {P : Type _} [EuclideanPlane P] (l : Seg_nd P) : (∃ q : P, l.1.target LiesInt (SEG l.1.source q)) := by 
+  let h := l.1.toVec +ᵥ l.1.target
+  let half : ℝ := 1/2
+  have c: 0 ≤ half ∧ half ≤ 1 ∧ VEC l.1.source l.1.target = half • VEC l.1.source h := by
+    norm_num
+    rw [seg_toVec_eq_vec]
+    repeat rw [Vec.mk_pt_pt]
+    rw [vadd_vsub_assoc]
+    let t := l.1.target -ᵥ l.1.source
+    have x: t = 1/2 * (t + t) := by
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean:304]: `field_simp` can solve this goal

File: EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean [hunk 6]
```diff
@@ -257,19 +292,70 @@ section existence
...
-
+theorem Seg_nd.exist_pt_beyond_pt {P : Type _} [EuclideanPlane P] (l : Seg_nd P) : (∃ q : P, l.1.target LiesInt (SEG l.1.source q)) := by 
+  let h := l.1.toVec +ᵥ l.1.target
+  let half : ℝ := 1/2
+  have c: 0 ≤ half ∧ half ≤ 1 ∧ VEC l.1.source l.1.target = half • VEC l.1.source h := by
+    norm_num
+    rw [seg_toVec_eq_vec]
+    repeat rw [Vec.mk_pt_pt]
+    rw [vadd_vsub_assoc]
+    let t := l.1.target -ᵥ l.1.source
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean:303]: Why don't use `l.1.toVec`?

File: EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean [hunk 7]
```diff
@@ -257,19 +292,70 @@ section existence
...
+  rcases h with ⟨⟨c, d⟩, b⟩
+  rcases b with ⟨p_ne_s, _⟩
+  rcases d with ⟨_, _, e⟩
+  have t: VEC Seg.source p ≠ 0 := by exact Iff.mp (ne_iff_vec_ne_zero Seg.source p) p_ne_s
+  have u: c • VEC Seg.source Seg.target ≠ 0 := by 
+    rw [← e]
+    exact t
+  have v: VEC Seg.source Seg.target ≠ 0 := by
+    exact right_ne_zero_of_smul u
+  exact Iff.mp vsub_ne_zero v
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean:339]: You may write this as `exact Iff.mp vsub_ne_zero (right_ne_zero_of_smul u)` to make prove shorter.

File: EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean [hunk 8]
```diff
@@ -257,19 +292,70 @@ section existence
...
 
 -- If a generalized directed segment contains an interior point, then it is nontrivial
-theorem nd_iff_exist_int_pt (l : Seg P) : ∃ (p : P), p LiesInt l ↔ l.is_nd := sorry
+theorem nd_iff_exist_int_pt (l : Seg P) : (∃ (p : P), p LiesInt l) ↔ l.is_nd := by
+  constructor
+
```
Comment (jjdishere) [inline EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean:344]: Please do not leave a blank line

**Review by jjdishere — COMMENTED** [review]
The proofs are good. Better still, you can make your proof more elegant. Would you please rewrite some of the proofs? This is not hard and the team leader can tell others about how to make proofs more elegant.

**Comment by jjdishere** [comment]
If you finished your change, please leave a comment here to ask for a new review. Otherwise I won't start to look at your changes. Thanks!

**Comment by kfc2333** [comment]
fixed build error and improved some proofs.

**Review by jjdishere — APPROVED** [review]
Great!

**Pull request merged**